### PR TITLE
Updates transAll route

### PR DESCRIPTION
**Pull request by warrenrhodes**

### CODE REVIEW

File: app_customers/lib/routes/app_router_gorouter.dart [hunk 1]
```diff
@@ -80,6 +96,7 @@ class _GoRouterRoutesProvider {
               buyerGatewayId: buyerGatewayId,
               featureReference: featureReference,
               receiverOperator: receiverOperator,
+              forfeitId: forfeitId == 'null' ? null : forfeitId,
```
Comment (nidovic) [inline app_customers/lib/routes/app_router_gorouter.dart:99]: can we avoid to put value 'null' ??

File: app_customers/lib/routes/pages_routes.dart [hunk 2]
```diff
@@ -97,35 +106,42 @@ class CreditTransactionParams {
...
-  final String buyerPhoneNumber;
+  final String? buyerPhoneNumber;
 
   /// The receiverNumber parameter.
-  final String receiverPhoneNumber;
+  final String? receiverPhoneNumber;
 
   /// The amountToPay parameter.
-  final String amountToPay;
+  final String? amountToPay;
```
Comment (nidovic) [inline app_customers/lib/routes/pages_routes.dart:125]:
```suggestion
  final String? amountInXAFToPay;
```

**Review by nidovic — APPROVED** [review]

File: app_customers/lib/routes/pages_routes.dart [hunk 3]
```diff
@@ -64,14 +68,19 @@ class _WelcomeRoute extends PagesRoutesNoParams {
...
+      '/init-transaction?${CreditTransactionParams.keyParamTransactionId}=${initCreditTransactionParams.transactionId}&${CreditTransactionParams.keyParamBuyerGatewayId}=${initCreditTransactionParams.buyerGatewayId}&${CreditTransactionParams.keyParamReceiverOperator}=${initCreditTransactionParams.receiverOperator}&${CreditTransactionParams.keyParamFeatureReference}=${initCreditTransactionParams.featureReference}&${CreditTransactionParams.keyParamBuyerPhoneNumber}=${initCreditTransactionParams.buyerPhoneNumber}&${CreditTransactionParams.keyParamReceiverPhoneNumber}=${initCreditTransactionParams.receiverPhoneNumber}&${CreditTransactionParams.keyParamAmountToPay}=${initCreditTransactionParams.amountToPay}&${CreditTransactionParams.keyParamForfeitId}=${initCreditTransactionParams.forfeitId}';
```
[the remark this answers is not in the document]
Comment (Tadjaur) [reply app_customers/lib/routes/pages_routes.dart:83]: Feel free to create a dedicated issue to address this one.

File: app_customers/lib/routes/app_router_gorouter.dart (hunk 1 above)
Comment (warrenrhodes) [reply app_customers/lib/routes/app_router_gorouter.dart:99]: nope when there is element depend on request we retrieve 'null' or null so it is very important  to have the 2 verification

**Pull request merged**